walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
+    int endDay = TimeRange.END_OF_DAY;
+
+    /*loop through all the planned events and find the relevant ones 
+    (where there's an attendee clash)
+    */
+    for (Event event: events) {
+    //get the attendees from the specific event you're checking the attendees of
+      Set<String> eventAttendees = event.getAttendees();
+      for (String attendee : attendeesList) {
+        if (eventAttendees.contains(attendee)) {
+          clashAttendeeEvents.add(event.getWhen());
+        }
+      }
+    }
+
+    /* loop through all events attended by this meeting attendees
+    to see if any of the times would clash
+    */
+    
+    // if there are no attendee-clash meetings, then the whole day is available
+    if (clashAttendeeEvents.size() == 0) {
+        if (meetingDuration <= TimeRange.WHOLE_DAY.duration()) {
+          proposedTimeRanges.add(TimeRange.WHOLE_DAY);
+        }
+      return (proposedTimeRanges);
+    }
+
+    // order the clashing events into start time order
+    Collections.sort(clashAttendeeEvents, TimeRange.ORDER_BY_START);
+    TimeRange startTime = TimeRange.fromStartEnd(startDay, clashAttendeeEvents.get(0).start(), false);
+
+    // only add the time if there's enough duration for the requested meeting
+    if (startTime.duration() >= request.getDuration() || Objects.equals(startTime.duration(), request.getDuration())){
+      proposedTimeRanges.add(startTime);
+    }   
+
+    // order by end time.
+    Collections.sort(clashAttendeeEvents, TimeRange.ORDER_BY_END);
+    TimeRange endTime = TimeRange.fromStartEnd(
+        clashAttendeeEvents.get(clashAttendeeEvents.size()-1).end(), endDay, true);
+        
+    if (!(endTime.duration() < request.getDuration())) {
+      proposedTimeRanges.add(endTime);
+    }
+
+    if (clashAttendeeEvents.size() == 1) {
+      return proposedTimeRanges;
+    }
+    
+    // For multiple events, there are the following cases:
+    // case 1: the events don't overlap and aren't contained
+    // case 2: the events overlap
+    // case 3: the event is contained by another
+
+    for (int i = 0; i < clashAttendeeEvents.size() - 1; i++) {
+      TimeRange currentEvent = clashAttendeeEvents.get(i);
+      TimeRange followingEvent = clashAttendeeEvents.get(i+1);
+
+      if (!(currentEvent.contains(followingEvent)) && !(currentEvent.overlaps(followingEvent))) {
+        // case 1
+        TimeRange gap = TimeRange.fromStartEnd(currentEvent.end(), followingEvent.start(), false);
+        long possibleDuration = gap.duration();
+        if (meetingDuration <= possibleDuration) {
+          proposedTimeRanges.add(gap);
+        }
+      } else if (!currentEvent.contains(followingEvent) && currentEvent.overlaps(followingEvent)) {
+        // case 2
+        // do nothing - not an eligible time
+      } else if (currentEvent.contains(followingEvent) && !currentEvent.overlaps(followingEvent)) {
+      // case 3
+      // do nothing - not an eligible time
+      }
```

Comment:
same here, please consider using `overlaps` only.
on a side note, it's great that you're highlighting the other cases here :) but as long as they are irrelevant and you do nothing with them, you can just have the comments without the `else if` statements